### ISSUE TITLE
ColoredFormatter can't handle unicode

### ISSUE THREAD

**Issue by zelial** [psss/python-nitrate#20]
When tcms-results tries to log info about bugs it processes, it tracebacks like this:

Traceback (most recent call last):
  File "/usr/lib64/python2.7/logging/__init__.py", line 851, in emit
    msg = self.format(record)
   File "/usr/lib64/python2.7/logging/__init__.py", line 724, in format
     return fmt.format(record)
   File "/usr/lib/python2.7/site-packages/nitrate/config.py", line 166, in format
     return u"{0} {1}".format(level, record.getMessage())
 UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 120: ordinal not in range(128)

**Comment by ZelenyMartin** [comment]
Hi, can you please elaborate more about how to reproduce the issue? I don't know what specific TR should I run the tcms-results on to get the traceback you have. Do you have already any hint? The 'u' string prefix can play some role here, but I have to check it first.

**Comment by ZelenyMartin** [comment]
So I have a reproducer:
```
import sys
from nitrate import log

log.info("test")
log.info(sys.argv[1])
```

**Comment by psss** [comment]
Seems like `tcms-results` is trying a log a string which contains unicode characters but is not properly encoded/decoded. I would say something like this is happening:

```python
# coding=utf8
x = '…'
print u'{}'.format(x)
```

It should be like this (which works correctly):

```python
# coding=utf8
x = u'…'
print u'{}'.format(x)
```

But removing the `u` prefix...

```python
# coding=utf8
x = u'…'
print '{}'.format(x)
```

... would result into another traceback when unicode provided:

```python
Traceback (most recent call last):
  File "ha.py", line 3, in <module>
    print '{}'.format(x)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2026' in position 0: ordinal not in range(128)
```

I guess the only way is to fix `tcms-results` to handle incoming `unicode` strings from Bugzilla properly.

**Comment by hroncok** [comment]
> This is issue only on python2. With python3, both variants are ok.

With python3, `sys.argv[1]` is an unicode text. On python2, it is a sequence of bytes.

**Comment by psss** [comment]
Seems the following should fix the issue:

```diff
-            log.info("BZ{0}".format(bug))
+            log.info(u"BZ{0}".format(bug))
```

**Comment by psss** [comment]
Should be fixed by #21. Seems to be working as expected:

```
>>> from nitrate import log
>>> log.setLevel(log.INFO)

>>> log.info('This is ascii.')
 INFO  This is ascii.

>>> log.info(u"This is encoded ůnicódé.")
 INFO  This is encoded ůnicódé.

>>> log.info("This is unencoded ůnicódé.")
 INFO  This is unencoded ůnicódé.

>>> log.info("This is weird content \xaf.")
 INFO  This is weird content �.
```

@ZelenyMartin, could you please verify with `tcms-results`?
